### ISSUE TITLE
Is there a filter to hide input on terminal?

### ISSUE THREAD

**Issue by francescobianca** [datafolklabs/cement#621]
I want to hidden the typing of this field:
It is possible in the latest version (3.0.6)? I didn't find anything about it in the doc

`password = shell.Prompt("Insert your password:")`

Thanks in advance!

**Comment by derks** [comment]
@francescobianca currently no. Actually surprised I hadn't thought to add support for it.  For this I'd used [getpass.getpass](https://docs.python.org/3/library/getpass.html) in the standard library.

```python
from getpass import getpass
password = getpass('Insert your password: ')
```

I'm going to keep this open, as it would be pretty trivial to add support for using `getpass` instead of `input`, and would be easily backward compatible.

**Comment by derks** [comment]
Notes for implementation: 

- Overriding the `input_func` would be most flexible, and would allow customization.
- Also abstract the input operation to `self._get_input(...)` so that can be sub-classed if desired.

```
from cement.utils import shell
from getpass import getpass

password = shell.Prompt('Enter password: ', input_func=getpass)
```

**Comment by francescobianca** [comment]
Hi @derks,

thanks! 
At the moment I have implemented as follows:
```python
from cement import Controller, ex, shell
from getpass import getpass

@ex(help='Login command')
def login(self):
    username = shell.Prompt("Enter username:")
    password = getpass("Enter password: ")
    # password = shell.Prompt("Insert your password:")

    authentication_service.login(username.input, password)
```

I will replace when the feature is added.
Great work, great framework!

**Comment by derks** [comment]
Great! I'll keep this open and linked for the next version.

**Comment by derks** [comment]
@francescobianca this will be available in stable/3.0.8:

```python
from cement.utils import shell

p = Prompt('Enter secure info:', suppress=True)
```
